gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientReadChannel.java

```diff
@@ -210,14 +212,41 @@ private class ContentReadChannel {
     // in-place seeks.
     private byte[] skipBuffer = null;
     private ReadableByteChannel byteChannel = null;
+    // Keeps track of distance between last 2 consecutive request.
+    private LimitedFifoQueue<Long> requestDistance = new LimitedFifoQueue<Long>(2);
```

Comment:
Should we create this only for AUTO_RANDOM?

Reply:
We can but it adds multiple if calls in regular path. Given it's limited to just 2 long I am not too much worried about it.

Reply:
Anyhow, it wasn't a big headache. Removed it and also made it configurable.